### ISSUE TITLE
XPU: aten.convolution_backward.default returns defined tensor even when output_mask=False

### ISSUE THREAD

**Issue by dayanandav** [pytorch/pytorch#172302]
### 🐛 Describe the bug

On Intel XPU (oneDNN backend),
aten::convolution_backward.default is return an empty tensor for an output(gradient) even when the corresponding output_mask entry is False.

This violates the ATen operator contract and causes downstream failures in DTensor, FakeTensor, torch.compile, and AOTAutograd.

***CPU and CUDA does not exhibit this behavior.

## Steps to Reproduce

```
import torch
device = 'xpu'
input = torch.randn(1, 1, 4, 4).to(device)
weight = torch.randn(1, 1, 3, 3).to(device)
bias = torch.randn(1).to(device)
grad_output = torch.randn(1, 1, 2, 2).to(device)
output_mask = [False, False, False]

grad_input, grad_weight, grad_bias = torch.ops.aten.convolution_backward(
	grad_output,            # grad_output
	input,                  # input
	weight,                 # weight
	[1],                    # bias_sizes
	(1, 1),                 # stride
	(0, 0),                 # padding
	(1, 1),                 # dilation
	False,                  # transposed
	(0, 0),                 # output_padding
	1,                      # groups
	output_mask,            # output_mask: (grad_input, grad_weight, grad_bias)
)
assert grad_input is None
assert grad_weight is None
assert grad_bias is None
```

### Versions

```
PyTorch version: 2.11.0a0+git6886c9d
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: SUSE Linux Enterprise Server 15 SP4 (x86_64)
GCC version: (GCC) 13.3.0
Clang version: Could not collect
CMake version: version 4.1.2
Libc version: glibc-2.31

Python version: 3.10.0 | packaged by conda-forge | (default, Nov 20 2021, 02:24:10) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.14.21-150400.24.55-default-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20250201
Intel GPU driver version:
* level-zero:   1.24.0.0-i1146
* intel-opencl: 25.18.33578.42-1146
Intel GPU models onboard:
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-1800-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [1] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-1800-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [2] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-4200-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [3] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-4200-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [4] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-6c00-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [5] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0000-6c00-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [6] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-1800-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [7] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-1800-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [8] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-4200-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [9] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-4200-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [10] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-6c00-000000000001, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
* [11] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBD6, uuid=8680d60b-2f00-0100-6c00-000000000002, driver_version='1.6.33578+38', total_memory=65520MB, max_compute_units=448, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   52 bits physical, 57 bits virtual
Byte Order:                      Little Endian
CPU(s):                          208
On-line CPU(s) list:             0-207
Vendor ID:                       GenuineIntel
Model name:                      Intel (R) Xeon (R) CPU Max 9470C
CPU family:                      6
Model:                           143
Thread(s) per core:              2
Core(s) per socket:              52
Socket(s):                       2
Stepping:                        8
Frequency boost:                 enabled
CPU max MHz:                     2001.0000
CPU min MHz:                     800.0000
BogoMIPS:                        4000.00
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr avx512_fp16 amx_tile flush_l1d arch_capabilities
Virtualization:                  VT-x
L1d cache:                       4.9 MiB (104 instances)
L1i cache:                       3.3 MiB (104 instances)
L2 cache:                        208 MiB (104 instances)
L3 cache:                        210 MiB (2 instances)
NUMA node(s):                    4
NUMA node0 CPU(s):               0-51,104-155
NUMA node1 CPU(s):               52-103,156-207
NUMA node2 CPU(s):
NUMA node3 CPU(s):
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] intel-cmplr-lib-ur==2025.3.1
[pip3] intel-openmp==2025.3.1
[pip3] mkl-include==2025.3.0
[pip3] mkl-static==2025.3.0
[pip3] numpy==2.2.6
[pip3] onemkl-license==2025.3.0
[pip3] optree==0.18.0
[pip3] pytorch-triton-xpu==3.5.0+git1b0418a9
[pip3] tbb==2022.3.0
[pip3] tbb-devel==2022.3.0
[pip3] tcmlib==1.4.1
[pip3] torch==2.11.0a0+git6886c9d
[pip3] torchdata==0.11.0
[pip3] umf==1.0.2
[conda] intel-cmplr-lib-ur         2025.3.1             pypi_0               pypi
[conda] intel-openmp               2025.3.1             pypi_0               pypi
[conda] mkl-include                2025.3.0             pypi_0               pypi
[conda] mkl-static                 2025.3.0             pypi_0               pypi
[conda] numpy                      2.2.6                py310hefbff90_0      conda-forge
[conda] onemkl-license             2025.3.0             pypi_0               pypi
[conda] optree                     0.18.0               pypi_0               pypi
[conda] pytorch-triton-xpu         3.5.0+git1b0418a9    pypi_0               pypi
[conda] tbb                        2022.3.0             pypi_0               pypi
[conda] tbb-devel                  2022.3.0             pypi_0               pypi
[conda] tcmlib                     1.4.1                pypi_0               pypi
[conda] torch                      2.11.0a0+git6886c9d  pypi_0               pypi
[conda] torchdata                  0.11.0               pypi_0               pypi
[conda] umf                        1.0.2                pypi_0               pypi

```

cc @gujinghui @EikanWang @fengyuan14 @guangyey
